yocs_cmd_vel_mux/include/yocs_cmd_vel_mux/cmd_vel_subscribers.hpp

```diff
@@ -62,25 +59,27 @@ class CmdVelSubscribers
   {
   public:
     unsigned int           idx;          /**< Index; assigned according to the order on YAML file */
-    std::string            name;         /**< Descriptive name */
-    ros::Subscriber        subs;         /**< The subscriber itself */
+    std::string            name;         /**< Descriptive name; must be unique to this subscriber */
     std::string            topic;        /**< The name of the topic */
+    ros::Subscriber        subs;         /**< The subscriber itself */
     ros::Timer             timer;        /**< No incoming messages timeout */
     double                 timeout;      /**< Timer's timeout, in seconds  */
     unsigned int           priority;     /**< UNIQUE integer from 0 (lowest priority) to MAX_INT */
     std::string            short_desc;   /**< Short description (optional) */
     bool                   active;       /**< Whether this source is active */
 
-    CmdVelSubs(unsigned int idx) : idx(idx), active(false) {};
+    CmdVelSubs(unsigned int idx) : idx(idx), active(false) { };
+    ~CmdVelSubs() { }
 
+    /** Fill attributes with a YAML node content */
     void operator << (const YAML::Node& node);
   };
 
   CmdVelSubscribers() : allowed(VACANT) { }
   ~CmdVelSubscribers() { }
 
-  std::vector<CmdVelSubs>::size_type size() { return list.size(); };
-  CmdVelSubs& operator [] (unsigned int idx) { return list[idx]; };
+  std::vector<std::shared_ptr<CmdVelSubs>>::size_type size() { return list.size(); };
+  std::shared_ptr<CmdVelSubs>& operator [] (unsigned int idx) { return list[idx]; };
```

Comment:
I'm fine with it as it is for now as it's only internally used...just leave a comment that it should ideally be updated to subclass the vector or pass out iterators instead of this reference specific functionality that we don't use.